src/validation/schema.js

```diff
@@ -380,15 +380,290 @@ const validateEntityFields = (defs, def) =>
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext 'language' value must be one of: simple, da, nl, en, fi, fr , de , hu , it , no , pt , ro , ru , es , sv , tr.`,
+      }),
+    ])
+  } else if (!languages.includes(argument.value.value)) {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext 'language' value must be one of: simple, da, nl, en, fi, fr , de , hu , it , no , pt , ro , ru , es , sv , tr.`,
+      }),
+    ])
+  } else {
+    return List([])
+  }
+}
+
+const validateFulltextAlgorithm = (def, directive) => {
+  let algorithm = directive.arguments.find(argument => argument.name.value == 'algorithm')
+  return algorithm
+    ? validateFulltextArgumentAlgorithm(def, directive, algorithm)
+    : List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'algorithm' must be specified`,
+        }),
+      ])
+}
+
+const validateFulltextArgumentAlgorithm = (def, directive, argument) => {
+  if (argument.value.kind != 'EnumValue') {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'algorithm' must be one of: rank, proximityRank`,
+      }),
+    ])
+  } else if (!['rank', 'proximity_rank'].includes(argument.value.value)) {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext 'algorithm' value, '${argument.value.value}', must be one of: rank, proximityRank`,
+      }),
+    ])
+  } else {
+    return List([])
+  }
+}
+
+const validateFulltextInclude = (def, directive) => {
+  let include = directive.arguments.find(argument => argument.name.value == 'include')
+  if (include) {
+    if (include.value.kind != 'ListValue') {
+      return List([
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: `@fulltext argument 'include' must be a list`,
+        }),
+      ])
+    }
+    return include.value.values.reduce(
+      (errors, type) =>
+        errors.concat(validateFulltextArgumentInclude(def, directive, type)),
+      List(),
+    )
+  } else {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'include' must be specified`,
+      }),
+    ])
+  }
+}
+
+const validateFulltextArgumentInclude = (def, directive, argument) => {
+  if (argument.kind != 'ObjectValue') {
+    return List([
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: `@fulltext argument 'include' must have the form '[{ entity: "entityName", fields: [{name: "fieldName"},...]}...]`,
```

Comment:
The whitespace in the example `[{ ... }]` code is used inconsistently (`{ entity:` vs. `[{name:` for instance). Also at the end there it's `...]}...]` instead of `... ]} ... ]`